### PR TITLE
adding CartoAsyncDatasource

### PR DESCRIPTION
**Pull request by juanrmn**

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -24,7 +24,7 @@ exclude_lines =
     if __name__ == .__main__.:
 
 ignore_errors = True
-fail_under = 85
```
Comment (juanrmn) [inline .coveragerc]: :-1:

File: pyproject.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name = "geographica-longitude"
 packages = [
     { include='longitude' }
 ]
-version = "1.0.0a1"
+version = "1.0.0a2"
```
Comment (josemazo) [inline pyproject.toml]: Take a look to this TOML file looking for pre-release things, and it will be better to modify the version to 0.9.0.

Comment (juanrmn) [reply pyproject.toml]: Done.

File: longitude/core/caches/redis.py [hunk 3]
```diff
@@ -34,5 +59,16 @@ def execute_put(self, key, payload, expiration_time_s=None):
             self._redis.expire(name=key, time=expiration_time_s)
```
[the remark this answers is not in the document]
Comment (juanrmn) [reply longitude/core/caches/redis.py]: Yes, it could, and it does it now.

File: longitude/core/caches/redis.py [hunk 4]
```diff
@@ -53,18 +53,24 @@ def execute_get(self, key):
 
     def execute_put(self, key, payload, expiration_time_s=None):
         overwrite = self._redis.exists(key) == 1
-        self._redis.set(name=key, value=payload)
+
+        opt = {}
         expiration_time_s = expiration_time_s or self.expiration_time
         if expiration_time_s:
-            self._redis.expire(name=key, time=expiration_time_s)
+            opt['ex'] = expiration_time_s
```
Comment (josemazo) [inline longitude/core/caches/redis.py]: :feelsgood:

**Pull request merged**